hvplot/interactive.py

```diff
@@ -512,3 +507,31 @@ def widgets(self):
                 if w not in widgets:
                     widgets.append(w)
         return pn.Column(*widgets)
+
+
+class _hvplot():
+    _converters = tuple(HoloViewsConverter._kind_mapping)
+
+    __slots__ = ["_interactive"]
+
+    def __init__(self, _interactive):
+        self._interactive = _interactive
+
+    def __call__(self, *args, _kind=None, **kwargs):
+        # The underscore in _kind is to be able to overwrite
+        # kind in kwargs, if used with partial. E.g. the following
+        # `dfi.hvplot.scatter('x','y', kind='area')` will return a
+        # scatter plot.
+        new = self._interactive._resolve_accessor()
+        transform = new._transform
+        transform = type(transform)(transform, 'hvplot', accessor=True)
+        if _kind:
+            kwargs["kind"] = _kind
+        dmap = 'kind' not in kwargs
```

Comment:
Not sure I follow this, why does not setting kind imply `dmap=True`?

Reply:
Ah, I see. I think the reasoning was that if `kind` is controlled by a widget or parameter then we should not use a `DynamicMap` to render the output because HoloViews cannot dynamically switch the plot type. So I suppose it should be:

```python
dmap = 'kind' not in kwargs or not isinstance(kwargs['kind'], str)
```